cmd/plugins_test.go

```diff
@@ -28,24 +31,61 @@ import (
 var initChan = make(chan struct{}, 256)
 var testDirRoot string
 
+const (
+	rootDir   string = "./"
+	prefixDir string = "plugin_test_tempdir"
+)
+
 // makeDirWithSharedObjects creates a new temporary directory containing files under the runtime directory
 // it compiles all .go files into shared object files with extension ext in the corresponding directory
 // It returns the root of the directory and a cleanup function.
-func makeDirWithSharedObjects(files map[string]string, ext string) (root string, cleanup func()) {
-	root, cleanup, err := test.MakeTempFS("./", "plugin_test_tempdir", files)
+func makeDirWithSharedObjects(files map[string]string, ext string) (tempRootDir string, cleanup func()) {
+	test.FuncEnter()
+	defer test.FuncExit()
+
+	tempRootDir, err := ioutil.TempDir(rootDir, prefixDir)
 	if err != nil {
 		panic(err)
 	}
+	cleanup = func() {
```

Comment:
I don't understand the reason for duplicating the implementation of `test.MakeTempFS` here. As long as we register the signal handler _before_ shelling out and running go build, this should fix the issue in the majority of cases.

Reply:
This was in the Github analysis I added. The temp directory uses a random generated suffix, therefore we do not know the directory to clean until MakeTempFS returns. And it only returns after directories, subdirectories and files are created. If a signal is received in the meantime these will not be cleaned. I managed to easily reproduce this condition and that is why the functionality was pulled up.

Reply:
Done, added pointer to explanation